### DISCUSSION TITLE
Fastlane for iOS: Missing password for user

### DISCUSSION THREAD

**Discussion by mregnauld** [codemagic-ci-cd/codemagic-docs#1395] [Q&A] [answered]
In my workflow (`63e6508991f4f35a9aaac6ed`), I try to upload screenshots to the App Store Connect.

But as soon as I call `fastlane screenshots`, I got the following error:

```
Missing password for user mregnauld@icloud.com, and running in non-interactive shell
```

which is actually not surprising, since I haven't set the password anywhere.

I also get the following message:

```
You can also pass the password using the `FASTLANE_PASSWORD` environment variable
```

but I don't really understand what to do exactly. I know how I can set a new environment variable named `FASTLANE_PASSWORD` in Codemagic, but I don't know what to do next...

Thanks.

**Comment by mregnauld** [comment]
OK so apparently, just setting the variable was enough.
But since two-factor authentication is enabled for my account, I'm still stuck because there is no way I can enter the 6 digit code...
How can I handle that?

**Comment by tiger5226** [comment]
you need to configure the API access as stated [here](https://docs.codemagic.io/integrations/fastlane-integration/). However, it does not tell you how to get the api key, but you can find detailed instructions on how to accomplish the creation [here](https://docs.codemagic.io/flutter-code-signing/ios-code-signing/) at step 2. Hope this helps.

**Reply by mregnauld** [reply]
I'll look at that, thanks.
But surprisingly, I don't have any problem when it comes to deploying the app to the App Store (but I directly use Codemagic without Fastlane in that case).

**Comment by mregnauld** [comment, marked as answer]
I finally managed to make it work using a JSON containing:
- my App Store Connect Key ID
- my Issuer ID
- my API Key (stored in the p8 file)

as explained here: https://docs.fastlane.tools/app-store-connect-api/
